crossdock/server/server.go

```diff
@@ -46,10 +48,11 @@ type Server struct {
 	Tracer           opentracing.Tracer
 	listener         net.Listener
 	channel          *tchannel.Channel
+	eHandler         *endtoend.Handler
 }
 
 // Start starts the test server called by the Client and other upstream servers.
-func (s *Server) Start() error {
+func (s *Server) Start(endToEndCfg config.Configuration) error {
```

Comment:
I need to inject it for testing purposes since the real localAgentHostPort will make the system panic. Is there a good way to set environment variables? If I can do that, they I wouldn't have to pass this around.